From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Fri, 4 Jun 2021 18:14:17 +0000
Subject: [PATCH] test: Fix declaration of test environment

meson 0.58.0 fails with:

    ERROR: List item must be one of <class 'str'>, not <class 'list'>
---
 test/automated/display/meson.build     | 3 +--
 test/automated/displayless/meson.build | 6 ++----
 test/meson.build                       | 3 +--
 3 files changed, 4 insertions(+), 8 deletions(-)

diff --git a/test/automated/display/meson.build b/test/automated/display/meson.build
index a7e2e0be2..ce0276ad9 100644
--- a/test/automated/display/meson.build
+++ b/test/automated/display/meson.build
@@ -13,8 +13,7 @@ foreach t: tests
   test(
     t[0],
     executable(t[0], t[1], dependencies: libnautilus_dep),
-    env: [
-      test_env,
+    env: test_env + [
       'G_TEST_BUILDDIR=@0@'.format(meson.current_build_dir()),
       'G_TEST_SRCDIR=@0@'.format(meson.current_source_dir())
     ],
diff --git a/test/automated/displayless/meson.build b/test/automated/displayless/meson.build
index 64b23bd2a..02f4f3309 100644
--- a/test/automated/displayless/meson.build
+++ b/test/automated/displayless/meson.build
@@ -45,27 +45,25 @@ foreach t: tests
   test(
     t[0],
     executable(t[0], t[1], files('test-utilities.c'), dependencies: libnautilus_dep),
-    env: [
-      test_env,
+    env: test_env + [
       'G_TEST_BUILDDIR=@0@'.format(meson.current_build_dir()),
       'G_TEST_SRCDIR=@0@'.format(meson.current_source_dir())
     ],
     timeout: 480
   )
 endforeach
 
 
 
 # Tests that read and write from the Tracker index are run using 'tracker-sandbox'
 # script to use a temporary instance of tracker-miner-fs instead of the session one.
 foreach t: tracker_tests
   test_exe = executable(t[0], t[1], files('test-utilities.c'), dependencies: libnautilus_dep)
   test(
     t[0],
     tracker_sandbox,
     args: ['--store-tmpdir', '--index-recursive-tmpdir', test_exe],
-    env: [
-      test_env,
+    env: test_env + [
       'G_TEST_BUILDDIR=@0@'.format(meson.current_build_dir()),
       'G_TEST_SRCDIR=@0@'.format(meson.current_source_dir())
     ],
diff --git a/test/meson.build b/test/meson.build
index ab588efd3..9f238aaad 100644
--- a/test/meson.build
+++ b/test/meson.build
@@ -4,8 +4,7 @@
 # test(
 #   name,
 #   executable,
-#   env: [
-#     test_env,
+#   env: test_env + [
 #     'G_TEST_BUILDDIR=@0@'.format(meson.current_build_dir()),
 #     'G_TEST_SRCDIR=@0@'.format(meson.current_source_dir())
 #   ]
